src/App.tsx

```diff
@@ -41,35 +54,113 @@ type State = {
   isStarted: boolean,
   isReversed: boolean,
   sortType: SortType,
+  wordsLength: number,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    isStarted: false,
+    isReversed: false,
+    sortType: SortType.NONE,
+    wordsLength: 1,
+  };
+
+  render() {
+    const {
+      isStarted,
+      isReversed,
+      sortType,
+      wordsLength,
+    } = this.state;
+    const changedArray
+      = getReorderedGoods(goodsFromServer, sortType, isReversed, wordsLength);
+
+    return (
+      <div className="App notification is-warning">
+        {!isStarted && (
+          <button
+            type="button"
+            className="button is-link"
+            onClick={() => this.setState({ isStarted: true })}
+          >
+            Start
+          </button>
+        )}
+
+        {isStarted && (
+          <>
+            <ul className="Goods">
+              {changedArray.map(goodItem => (
+                <li className="Goods__item" key={uuid4()}>
+                  {goodItem}
+                </li>
+              ))}
+            </ul>
+
+            <div className="options">
+              <button
+                type="button"
+                className="button is-primary"
+                onClick={() => {
+                  this.setState({ sortType: SortType.ALPHABET });
+                }}
+              >
+                Sort alphabetically
+              </button>
+
+              <button
+                type="button"
+                className="button is-primary"
+                onClick={() => {
+                  this.setState({ sortType: SortType.LENGTH });
+                }}
+              >
+                Sort by length
+              </button>
+
+              <button
+                type="button"
+                className="button is-info"
+                onClick={() => {
+                  this.setState((state) => ({ isReversed: !state.isReversed }));
+                }}
+              >
+                Reverse
+              </button>
+
+              <button
+                type="button"
+                className="button is-danger"
+                onClick={() => {
+                  this.setState({
+                    sortType: SortType.NONE,
+                    isReversed: false,
+                    wordsLength: 1,
+                  });
```

Comment:
It is better to create methods and functions separate and later call them from your element